crates/nu-protocol/src/deprecation.rs

```diff
@@ -0,0 +1,28 @@
+use crate::{ParseWarning, ast::Call};
+
+#[derive(Clone)]
+pub enum DeprecationStatus {
+    Undeprecated,
+    Deprecated(Option<String>),
+}
```

Comment:
I don't like this enum as it is. `Undeprecated` sounds weird to me, I think we could better work with an `Option` instead. Also only having `Deprecated` is not granular enough. We talked about different deprecation levels, this doesn't provide that.

I'm also thinking about stronger deprecation entries, they could include:
- deprecation level, how often to notify the user
- since when it is deprecated, a version maybe
- expected removal if even
- reason for it
- help not

Only having a string incentivizes to weak messaging.

Reply:
I added all of these except "reason for it", since I figured that would go in the help string and it's not really clear where else that would fit into the `miette` error